src/Microsoft.SqlTools.ServiceLayer/AzureFunctions/InsertSqlBindingOperation.cs

```diff
@@ -0,0 +1,143 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.SqlTools.ServiceLayer.AzureFunctions.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+using Microsoft.SqlTools.Utility;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.SqlTools.ServiceLayer.AzureFunctions
+{
+    /// <summary>
+    /// Class to represent inserting a sql binding into an Azure Function
+    /// </summary>
+    class InsertSqlBindingOperation
+    {
+        const string functionAttributeText = "FunctionName";
+
+        public InsertSqlBindingParams Parameters { get; }
+
+        public InsertSqlBindingOperation(InsertSqlBindingParams parameters)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        public ResultStatus AddBinding()
+        {
+            try
+            {
+                string text = File.ReadAllText(Parameters.filePath);
+
+                SyntaxTree tree = CSharpSyntaxTree.ParseText(text);
+                CompilationUnitSyntax root = tree.GetCompilationUnitRoot();
+
+                // look for Azure Function to update
+                IEnumerable<MethodDeclarationSyntax> azureFunctionMethods = from methodDeclaration in root.DescendantNodes().OfType<MethodDeclarationSyntax>()
+                                                                            where methodDeclaration.AttributeLists.Count > 0
+                                                                            where methodDeclaration.AttributeLists.Where(a => a.Attributes.Where(attr => attr.Name.ToString().Contains(functionAttributeText) && attr.ArgumentList.Arguments.First().ToString().Equals($"\"{Parameters.functionName}\"")).Count() == 1).Count() == 1
+                                                                            select methodDeclaration;
+
+                if (azureFunctionMethods.Count() == 0)
+                {
+                    return new ResultStatus()
+                    {
+                        Success = false,
+                        ErrorMessage = $"Couldn't find Azure function with FunctionName {Parameters.functionName} in {Parameters.filePath}"
+                    };
+                }
+                else if (azureFunctionMethods.Count() > 1)
+                {
+                    return new ResultStatus()
+                    {
+                        Success = false,
+                        ErrorMessage = $"More than one Azure function found with the FunctionName {Parameters.functionName} in {Parameters.filePath}"
```

Comment:
are we using fully qualifies names like namespace.class.function or just the name? any pros and cons on using one or the other...?

Reply:
Just the name specified in the `[FunctionName("exampleName")] `attribute. There's actually a build error if more than one functions have the same value for FunctionNameAttribute 
![image](https://user-images.githubusercontent.com/31145923/128095434-a9166af8-07d6-4e27-a9a3-d411eac2608d.png)